<a href="https://colab.research.google.com/github/rknethinti/EDA-Airbnb-bookings-analysis/blob/main/Airbnb_Bookings_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Team Name: Kaimur Data Stimulators**

**Team Members:** 1) Nethinti Ramakrishna (Team Leader)
              2) Sahil Sukhadeve
              3) Ghanshyam Mali
              4) Sudhanshu Patil

## **What is Airbnb?**
#### Airbnb, Inc. is an American company that operates an online marketplace for lodging, primarily homestays for vacation rentals, and tourism activities. Based in San Francisco, California, the platform is accessible via website and mobile app. Airbnb does not own any of the listed properties; instead, it profits by receiving commission from each booking. The company was founded in 2008 by Brian Chesky, Nathan Blecharczyk and Joe Gebbia. Airbnb is a shortened version of its original name, AirBedandBreakfast.com.

# **Our Objective in this Analysis::**



### <b> Since 2008, guests and hosts have used Airbnb to expand on traveling possibilities and present a more unique, personalized way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analyzed and used for security, business decisions, understanding of customers' and providers' (hosts) behavior and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more. </b>

### <b>This dataset has around 49,000 observations in it with 16 columns and it is a mix between categorical and numeric values. </b>

### <b> Explore and analyze the data to discover key understandings (not limited to these) such as : 
* What can we learn about different hosts and areas?
* What can we learn from predictions? (ex: locations, prices, reviews, etc)
* Which hosts are the busiest and why?
* Is there any noticeable difference of traffic among different areas and what could be the reason for it? </b>

In [4]:
# importing important libraries using in this EDA
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
path= "/content/drive/MyDrive/Alma Better/Modules/Topic/Capstone Projects/Airbnb Bookings analysis- Nethinti Ramakrishna/Airbnb NYC 2019.csv"

In [5]:
airbnb= pd.read_csv(path)

In [6]:
airbnb.head()

id  ... availability_365
0  2539  ...              365
1  2595  ...              355
2  3647  ...              365
3  3831  ...              194
4  5022  ...                0

[5 rows x 16 columns]

In [ ]:
airbnb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [ ]:
airbnb.describe()

id       host_id      latitude     longitude         price  \
count  4.889500e+04  4.889500e+04  48895.000000  48895.000000  48895.000000   
mean   1.901714e+07  6.762001e+07     40.728949    -73.952170    152.720687   
std    1.098311e+07  7.861097e+07      0.054530      0.046157    240.154170   
min    2.539000e+03  2.438000e+03     40.499790    -74.244420      0.000000   
25%    9.471945e+06  7.822033e+06     40.690100    -73.983070     69.000000   
50%    1.967728e+07  3.079382e+07     40.723070    -73.955680    106.000000   
75%    2.915218e+07  1.074344e+08     40.763115    -73.936275    175.000000   
max    3.648724e+07  2.743213e+08     40.913060    -73.712990  10000.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    48895.000000       48895.000000       38843.000000   
mean         7.029962          23.274466           1.373221   
std         20.510550          44.550582           1.680442   
min          1.000000           0.000000           0.010000   
25%          1.000000           1.000000           0.190000   
50%          3.000000           5.000000           0.720000   
75%          5.000000          24.000000           2.020000   
max       1250.000000         629.000000          58.500000   

       calculated_host_listings_count  availability_365  
count                    48895.000000      48895.000000  
mean                         7.143982        112.781327  
std                         32.952519        131.622289  
min                          1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000         45.000000  
75%                          2.000000        227.000000  
max                        327.000000        365.000000

In [ ]:
airbnb.isnull().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

as "last_review" and "reviews_per_month" have more than 10,000 null values, it affects the outcomes of Data analysis; So, we are
removing these columns and also as we are not doing any analysis specifically on latitude and longitude, we're also removing these 
variables as well

In [ ]:
# dropping unnecessary data

airbnb.drop(['last_review','reviews_per_month','latitude','longitude'],axis=1,inplace=True)

In [ ]:
airbnb.head(2)

id                                name  host_id host_name  \
0  2539  Clean & quiet apt home by the park     2787      John   
1  2595               Skylit Midtown Castle     2845  Jennifer   

  neighbourhood_group neighbourhood        room_type  price  minimum_nights  \
0            Brooklyn    Kensington     Private room    149               1   
1           Manhattan       Midtown  Entire home/apt    225               1   

   number_of_reviews  calculated_host_listings_count  availability_365  
0                  9                               6               365  
1                 45                               2               355

In [ ]:
airbnb.shape

(48895, 12)

Now we have 48895 rows and 12 important columns ready for analysis

In [ ]:
list(airbnb.columns)

['id',
 'name',
 'host_id',
 'host_name',
 'neighbourhood_group',
 'neighbourhood',
 'room_type',
 'price',
 'minimum_nights',
 'number_of_reviews',
 'calculated_host_listings_count',
 'availability_365']

# 1) What can we learn about different hosts and areas?

In [ ]:
# analysing hosts and areas using groupby function

hosts_and_areas= airbnb.groupby(['host_name','neighbourhood_group'])["calculated_host_listings_count"].count().reset_index()

hosts_and_areas.sort_values('calculated_host_listings_count',ascending=False).head(10)

host_name neighbourhood_group  calculated_host_listings_count
13217  Sonder (NYC)           Manhattan                             327
1834     Blueground           Manhattan                             230
9742        Michael           Manhattan                             212
3250          David           Manhattan                             202
9741        Michael            Brooklyn                             159
6808           John           Manhattan                             151
3249          David            Brooklyn                             142
7275           Kara           Manhattan                             135
432            Alex           Manhattan                             134
9856           Mike           Manhattan                             134

### Here are some intersting findings,we found fascinating: 

#### 8 out of top 10 hosts are from the  ''Manhattan''  neighbourhood group.

#### 2 out of top 10 hosts are from the  ''Brooklyn"  neighbourhood group.

#### [ Sonder (NYC), Blueground, Michael, David, John, Kara, Alex, Mike] are the top hosts of   Manhattan neighbourhood group.

#### [ Michael,David] are the top hosts in Brooklyn neighbourhood group.

#### Manhattan neighbourhood group hosts are out-performing in listings.


# 2) What can we learn from predictions? (ex: locations, prices, reviews, etc)

In [ ]:
# prices in different localities
loc_price= airbnb.groupby(['name','neighbourhood_group','neighbourhood','price','minimum_nights'])['number_of_reviews'].count().reset_index()

top_prices= loc_price.sort_values('price',ascending=False).head(20)

In [ ]:
top_prices

name neighbourhood_group  \
20222                Furnished room in Astoria apartment              Queens   
1112                                 1-BR Lincoln Center           Manhattan   
27228    Luxury 1 bedroom apt. -stunning Manhattan views            Brooklyn   
36159                Quiet, Clean, Lit @ LES & Chinatown           Manhattan   
41053                                 Spanish Harlem Apt           Manhattan   
2225   2br - The Heart of NYC: Manhattans Lower East ...           Manhattan   
7092   Beautiful/Spacious 1 bed luxury flat-TriBeCa/Soho           Manhattan   
19735                                      Film Location            Brooklyn   
18366          East 72nd Townhouse by (Hidden by Airbnb)           Manhattan   
2749                 70' Luxury MotorYacht on the Hudson           Manhattan   
20464                               Gem of east Flatbush            Brooklyn   
2451                    3000 sq ft daylight photo studio           Manhattan   
4205                Apartment New York \nHell’s Kitchens           Manhattan   
37639                   SUPER BOWL Brooklyn Duplex Apt!!            Brooklyn   
27697       Luxury TriBeCa Apartment at an amazing price           Manhattan   
32010          Park Avenue Mansion by (Hidden by Airbnb)           Manhattan   
27861                 Luxury townhouse Greenwich Village           Manhattan   
45627                 UWS 1BR w/backyard + block from CP           Manhattan   
44128                SuperBowl Penthouse Loft 3,000 sqft           Manhattan   
28699    Midtown Manhattan great location (Gramacy park)           Manhattan   

           neighbourhood  price  minimum_nights  number_of_reviews  
20222            Astoria  10000             100                  1  
1112     Upper West Side  10000              30                  1  
27228         Greenpoint  10000               5                  1  
36159    Lower East Side   9999              99                  1  
41053        East Harlem   9999               5                  1  
2225     Lower East Side   9999              30                  1  
7092             Tribeca   8500              30                  1  
19735       Clinton Hill   8000               1                  1  
18366    Upper East Side   7703               1                  1  
2749   Battery Park City   7500               1                  1  
20464      East Flatbush   7500               1                  1  
2451             Chelsea   6800               1                  1  
4205     Upper West Side   6500              30                  1  
37639       Clinton Hill   6500               1                  1  
27697            Tribeca   6500             180                  1  
32010    Upper East Side   6419               1                  1  
27861  Greenwich Village   6000               1                  1  
45627    Upper West Side   6000              14                  1  
44128       Little Italy   5250               1                  1  
28699            Midtown   5100              30                  1

## Price Vs Location analysis shown some interesting facts:

###  The highest price is 10,000 usd and can be seen in  Astoria, Upper west side, lower east side, Greenpoint and East Harlem.

#### most minimum nights  spent in top 20 price list are,

###### - Luxury TriBeCa Apartment at an amazing price - 180 nights
###### - Furnished room in Astoria apartment - 100 nights
###### - Quiet, Clean, Lit @ LES & Chinatown	- 99 nights
###### - highest priced rooms (i.e., 10,000usd) are present in Manhattan, Brooklyn and Queens neighbourhood group 



# 3) Which hosts are the busiest and why?

In [ ]:
# finding busiest hosts 

busiest_hosts= airbnb.groupby(['host_name','neighbourhood_group','room_type'])["minimum_nights"].count().reset_index()

busiest_hosts.sort_values('minimum_nights', ascending=False).head()

host_name neighbourhood_group        room_type  minimum_nights
16549  Sonder (NYC)           Manhattan  Entire home/apt             319
2295     Blueground           Manhattan  Entire home/apt             230
12299       Michael           Manhattan  Entire home/apt             143
9190           Kara           Manhattan  Entire home/apt             129
4128          David           Manhattan  Entire home/apt             117

#### 5 out of top 5 are all from " Manhattan" neighbourhood group

#### Sonder(NYC) , Blueground, Michael, Kara, David are the top 5 most busiest hosts

# 4) Is there any noticeable difference of traffic among different areas and what could be the reason for it?

In [ ]:
traffic_areas= airbnb.groupby(['neighbourhood_group','room_type'])['minimum_nights'].count().reset_index()
traffic_areas.sort_values('minimum_nights',ascending=False).head(15)

neighbourhood_group        room_type  minimum_nights
6            Manhattan  Entire home/apt           13199
4             Brooklyn     Private room           10132
3             Brooklyn  Entire home/apt            9559
7            Manhattan     Private room            7982
10              Queens     Private room            3372
9               Queens  Entire home/apt            2096
1                Bronx     Private room             652
8            Manhattan      Shared room             480
5             Brooklyn      Shared room             413
0                Bronx  Entire home/apt             379
11              Queens      Shared room             198
13       Staten Island     Private room             188
12       Staten Island  Entire home/apt             176
2                Bronx      Shared room              60
14       Staten Island      Shared room               9